### PR TITLE
ARFC Moltres website migration

### LINKED ISSUE #179: Set up a website with documentation based on the MooseDocs system
We currently rely on Doxygen for Moltres' documentation on [https://arfc.github.io/moltres/](https://arfc.github.io/moltres/). Ideally, the documentation should contain helpful information about the input parameters of each class and sample input files showing how they're used.

While Doxygen documentation provides useful information on class inheritance trees, functions, and attributes, it does not provide any information on the input parameters of each class because MOOSE handles input parameters through instantiations of `InputParameters`.

The `MooseDocs` system is a potential solution as it provides a framework for generating a customizable website and documentation for classes as shown in [here](https://mooseframework.inl.gov/source/index.html). MOOSE uses this system for their main website. Instructions for setting it up can be found [here](https://mooseframework.inl.gov/python/MooseDocs/index.html).

This issue can be closed when a website is set up for Moltres. The website should at minimum include a homepage, a MooseDocs documentation page, and a Doxygen page, with proper links set up for user-friendliness.

### PR DESCRIPTION
**Pull request by smpark7**
Fixes #179
## Description
This PR migrates existing content on the ARFC Moltres website ([https://arfc.github.io/software/moltres/](https://arfc.github.io/software/moltres/)) to the newer MooseDocs-based website ([https://arfc.github.io/moltres/](https://arfc.github.io/moltres/)). A preview with the changes in this PR is available on my personal github website ([https://smpark7.github.io/moltres/index.html](https://smpark7.github.io/moltres/index.html)).

## Changes
- Migrated existing website content
  - Input syntax guide
  - Corresponding terms in the PDEs for each Kernel/BC object
  - Recommended executioner and preconditioner settings guide
- Added more content
  - Additional kernel/BC documentation: The weak form for second-order differential terms which were integrated by parts
  - Gwen's dissertation entry under "Publications"
- Tutorials
  - Moved and edited group constant generation tutorial README instructions into tutorial guides on the new Moltres website
  - Added input files for eigenvalue and transient tutorial problems
  - Added syntax check tests for these new tutorial input files
- Fixes
  - Typo in Publications webpage
  - Improve visibility of dropdown menu in dark mode
  - Removed redundant entry in godiva_serpent.inp

## Impact
- More extensive Moltres documentation support with the MooseDocs-based website
- Powerful MooseDocs-based tools which help with website maintenance

### CODE REVIEW

**Review by abachma2 — DISMISSED** [review]
Hi @smpark7. Great work on this, it looks really good overall. I had a few very small suggestions that are optional. I saw that one of the CI tests fails and I didn't really dig into why. I know you have a few PRs right now, so I'm guessing one of those PRs fixes it all. Let me know when this is ready to merge.

File: doc/content/getting_started/transient.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Transient Simulation with Temperature Coupling and Precursor Drift
+
+TODO: Add content
```
[the remark this answers is not in the document]
Comment (smpark7) [reply doc/content/getting_started/transient.md:3]: Good point! I'll make these changes.

**Review by abachma2 — DISMISSED** [review]

**Comment by smpark7** [comment]
The tests are running properly now. I just added an input mesh for the tutorial which I initially missed because of gitignore. This PR is ready for review and merge (unless you have new comments). @abachma2

**Comment by smpark7** [comment]
@abachma2 pinging you for a review and merge

**Review by abachma2 — APPROVED** [review]

**Pull request merged**